packages/events/src/service/events/reindex.test.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import { TRPCError } from '@trpc/server'
+import { http, HttpResponse } from 'msw'
+import {
+  ActionStatus,
+  ActionType,
+  EventDocument,
+  generateEventDocument,
+  getUUID,
+  SCOPES,
+  TENNIS_CLUB_MEMBERSHIP
+} from '@opencrvs/commons'
+import { tennisClubMembershipEvent } from '@opencrvs/commons/fixtures'
+import { createSystemTestClient, setupTestCase } from '@events/tests/utils'
+import {
+  getEventIndexName,
+  getOrCreateClient
+} from '@events/storage/elasticsearch'
+import { mswServer } from '@events/tests/msw'
+import { env } from '@events/environment'
+
+const spy = vi.fn()
+
+const postHandler = http.post(
+  `${env.COUNTRY_CONFIG_URL}/reindex`,
+  async (req) => {
+    const body = await req.request.json()
+    spy(body)
+
+    return HttpResponse.json(
+      {},
+      // @ts-expect-error - "For some reason the msw types here complain about the status, even though this is correct"
+      { status: 200 }
+    )
+  }
+)
+
+let event: EventDocument
+
+beforeEach(async () => {
+  mswServer.use(postHandler)
+  spy.mockReset()
+  const { user, eventsDb } = await setupTestCase()
+
+  event = generateEventDocument({
+    user,
+    configuration: tennisClubMembershipEvent,
+    actions: [ActionType.CREATE, ActionType.DECLARE]
+  })
+
+  const eventInDb = await eventsDb
+    .insertInto('events')
+    .values({
+      eventType: event.type,
+      transactionId: getUUID(),
+      trackingId: event.trackingId,
+      createdAt: event.createdAt,
+      updatedAt: event.updatedAt
+    })
+    .returning('id')
+    .executeTakeFirst()
```

Comment:
don't care here but fyi `executeTakeFirstOrThrow` should work without having to non-null assert